tools/bin/commands/util/operator_funcs

```diff
@@ -101,7 +101,7 @@ build_operator_local() {
         echo "Running tests"
         go test $TEST_TARGETS
     fi
-    if [ "$do_linux" ] && [ $(isOSX) ]; then
+    if [ "$do_linux" ] && isMacOs; then
```

Comment:
```suggestion
    if [ "$do_linux" ] && $(isMacOs); then
```

Reply:
@zregvart why ?

isMacOs returns either "true" or "false" which are valid Unix commands and get executed.

Reply:
$(isMacOs) would fire also for "false" (as this is a non-0 length string)

Reply:
I know, we used it differently, but I'd like to change to this kind of usage for boolean checks as it more clear than requiring to return "true" (or anything larger than 1 char) for true, and "" for false (has it is done in hasOption method). But this would be a larger change.

In fact I would even go for changing to getopts instead of our own grown option checks.

Reply:
And finally, I would rewrite in go (if people would feel comfortable with go)

Reply:
> @zregvart why ?

I could be misunderstanding something, based on:

```shell
wat() {
  echo "false"
}

if [ "true" ] && wat; then
  echo 1
fi

if [ "true" ] && $(wat); then
  echo 2
fi
```

Outputs:
```
false
1
```

I would expect it only to output:
```
false
```

Reply:
Yes, you shoudl use true and false without quote.

```
if  false ....
```

that's its how it is interpolaed by the shell when you call 

```
if isMacOs; then ..
```

(note the missing test operator [ ... ] !)

Reply:
sorry, just re-read you sample. trying now ...

Reply:
you are right. I have to eval it, but without test operator.